### DISCUSSION TITLE
Getting the direction of a COM port

### DISCUSSION THREAD

**Discussion by PeterStaev** [serialport/node-serialport#2673] [Q&A]
I'm lookgin for a way to get the direction of a COM port. For example in the situation below I want to list just the outbound port:
![image](https://github.com/serialport/node-serialport/assets/9464097/a8ae303c-dbed-4900-8e7c-4b15b30300f4)

Is there anyway to achieve this?

**Comment by GazHank** [comment]
I think you need to use the name of the port to identify the directionality.  In your case the one with the extra text at the end is outbound.  That seems to be what others have done with similar devices https://stackoverflow.com/questions/52327458/get-bluetooth-com-ports?rq=1

**Comment by PeterStaev** [comment]
@GazHank , sorry for my late reply and thanks for your suggestion! I want to even pull and show the actual name, but the problem is I do not see a way to do this with `SerialPort`. Here is what the `.list()` result shows for the same machine from which is the screenshot above:
![image](https://github.com/serialport/node-serialport/assets/9464097/f1656161-becd-4b98-8b53-b0230b9e7927)
